--- a/common/rdr/AESInStream.cxx	2026-08-21 14:19:44.682972527 +0000
+++ b/common/rdr/AESInStream.cxx	2026-08-21 14:19:44.684822109 +0000
@@ -27,6 +27,9 @@
 #include <rdr/AESInStream.h>
 
 #ifdef HAVE_NETTLE
+
+#include <nettle/version.h>
+
 using namespace rdr;
 
 AESInStream::AESInStream(InStream* _in, const uint8_t* key,
@@ -62,12 +65,20 @@
     EAX_SET_NONCE(&eaxCtx128, aes128_encrypt, 16, counter);
     EAX_UPDATE(&eaxCtx128, aes128_encrypt, 2, ad);
     EAX_DECRYPT(&eaxCtx128, aes128_encrypt, length, (uint8_t*)end, data);
+#if NETTLE_VERSION_MAJOR >= 4
+    EAX_DIGEST(&eaxCtx128, aes128_encrypt, macComputed);
+#else
     EAX_DIGEST(&eaxCtx128, aes128_encrypt, 16, macComputed);
+#endif
   } else {
     EAX_SET_NONCE(&eaxCtx256, aes256_encrypt, 16, counter);
     EAX_UPDATE(&eaxCtx256, aes256_encrypt, 2, ad);
     EAX_DECRYPT(&eaxCtx256, aes256_encrypt, length, (uint8_t*)end, data);
+#if NETTLE_VERSION_MAJOR >= 4
+    EAX_DIGEST(&eaxCtx256, aes256_encrypt, macComputed);
+#else
     EAX_DIGEST(&eaxCtx256, aes256_encrypt, 16, macComputed);
+#endif
   }
   if (memcmp(mac, macComputed, 16) != 0)
     throw std::runtime_error("AESInStream: Failed to authenticate message");
--- a/common/rdr/AESOutStream.cxx	2026-08-21 14:19:44.687159476 +0000
+++ b/common/rdr/AESOutStream.cxx	2026-08-21 14:19:44.688730313 +0000
@@ -27,6 +27,9 @@
 #include <rdr/AESOutStream.h>
 
 #ifdef HAVE_NETTLE
+
+#include <nettle/version.h>
+
 using namespace rdr;
 
 const int MaxMessageSize = 8192;
@@ -83,12 +86,20 @@
     EAX_SET_NONCE(&eaxCtx128, aes128_encrypt, 16, counter);
     EAX_UPDATE(&eaxCtx128, aes128_encrypt, 2, msg);
     EAX_ENCRYPT(&eaxCtx128, aes128_encrypt, length, msg + 2, data);
+#if NETTLE_VERSION_MAJOR >= 4
+    EAX_DIGEST(&eaxCtx128, aes128_encrypt, msg + 2 + length);
+#else
     EAX_DIGEST(&eaxCtx128, aes128_encrypt, 16, msg + 2 + length);
+#endif
   } else {
     EAX_SET_NONCE(&eaxCtx256, aes256_encrypt, 16, counter);
     EAX_UPDATE(&eaxCtx256, aes256_encrypt, 2, msg);
     EAX_ENCRYPT(&eaxCtx256, aes256_encrypt, length, msg + 2, data);
+#if NETTLE_VERSION_MAJOR >= 4
+    EAX_DIGEST(&eaxCtx256, aes256_encrypt, msg + 2 + length);
+#else
     EAX_DIGEST(&eaxCtx256, aes256_encrypt, 16, msg + 2 + length);
+#endif
   }
   out->writeBytes(msg, 2 + length + 16);
   out->flush();