--- a/common/rdr/FdInStream.cxx	2026-08-24 08:00:00.000000000 +0000
+++ b/common/rdr/FdInStream.cxx	2026-08-24 08:00:00.000000000 +0000
@@ -27,7 +27,6 @@
 #ifdef _WIN32
 #include <winsock2.h>
 #define errorNumber WSAGetLastError()
-#define close closesocket
 #include <core/winerrno.h>
 #else
 #include <sys/types.h>
@@ -54,7 +53,13 @@

 FdInStream::~FdInStream()
 {
-  if (closeWhenDone) close(fd);
+  if (closeWhenDone) {
+#ifdef _WIN32
+    closesocket(fd);
+#else
+    close(fd);
+#endif
+  }
 }


